### PR TITLE
feat: support xterm render type select

### PR DESCRIPTION
**Pull request by life2015**
### Types

支持切换 Xterm 的渲染模式： WebGL / Canvas / DOM

另外修改 TypeAhead 的不同渲染模式设置到枚举模式

- [x] 🎉 New Features

### <samp>🤖 Generated by Copilot at 1073207</samp>

*  Add xterm render type preference to terminal module
  * Define localization keys for preference name and description in English and Chinese languages ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-f66d683d6cd38ba54412c1c0a107842d6d65eb4312f6980a82ad912c4d51d9f4R474-R476), [link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-a5b8f7c1565e0628de0e7e6586f3ec3457c5e82522354bbd3f58926ecd3cdeb7R455-R456))
  * Define preference id in `CodeTerminalSettingId` enum in `packages/terminal-next/src/common/preference.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c1ed5d78a386f847107e56bdfa1704981305b09002593d3285e27eabbd50d8dcR124))
  * Define preference schema with description, constraint, and default value in `packages/terminal-next/src/common/preference.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c1ed5d78a386f847107e56bdfa1704981305b09002593d3285e27eabbd50d8dcR338-R343))
  * Register preference in preference settings service in `packages/preferences/src/browser/preference-settings.service.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-7a72db174865394cd9a826feb5fe2092d204e13ef8333f83f7c6a5edac12a70cR811))
  * Import preference service and preference id in `packages/terminal-next/src/browser/xterm.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36R10), [link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36L16-R17))
  * Inject preference service as a dependency in `XTerm` class using `Autowired` decorator in `packages/terminal-next/src/browser/xterm.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36L51-R57))
  * Change visibility of `enableCanvasRenderer` and `enableWebglRenderer` methods from private to protected in `XTerm` class in `packages/terminal-next/src/browser/xterm.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36L79-R83), [link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36L102-R102))
  * Add logic to get preference value and call corresponding enable method in `open` method of `XTerm` class in `packages/terminal-next/src/browser/xterm.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c0fa7b9d8ca7f7ccc298f511357563f0e5957a84ad18b44077bcef2e79726e36L192-R199))
  * Define enum values for valid cursor styles in `packages/terminal-next/src/common/preference.ts` ([link](https://github.com/opensumi/core/pull/2754/files?diff=unified&w=0#diff-c1ed5d78a386f847107e56bdfa1704981305b09002593d3285e27eabbd50d8dcR329))

### <samp>🤖 Generated by Copilot at 1073207</samp>

This pull request adds a new user preference for the terminal module to choose the xterm render type, which can be either `canvas` or `dom`. It also updates the preference schema, the localization files, and the preference settings service to support this new option. The changes affect the files `xterm.ts`, `preference.ts`, `en-US.lang.ts`, `zh-CN.lang.ts`, and `preference-settings.service.ts`.

### CODE REVIEW

File: packages/terminal-next/src/browser/xterm.ts [hunk 1]
```diff
@@ -189,8 +189,14 @@ export class XTerm extends Disposable implements IXTerm {
 
   open() {
     this.raw.open(this.container);
-    if (this.loadWebGLAddon()) {
+    const renderType = this.preferenceService.get<'canvas' | 'webgl' | 'dom'>(
```
Comment (erha19) [inline packages/terminal-next/src/browser/xterm.ts:192]: 这里的 type 感觉也可以用个枚举值来表示类型及值

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/terminal-next/src/browser/xterm.ts (hunk 1 above)
Comment (life2015) [reply packages/terminal-next/src/browser/xterm.ts]: 已修改，这部分都用 enum 了

File: packages/terminal-next/src/common/preference.ts [hunk 2]
```diff
@@ -333,5 +335,11 @@ export const terminalPreferenceSchema: PreferenceSchema = {
       enum: [TerminalCursorStyle.BLOCK, TerminalCursorStyle.LINE, TerminalCursorStyle.UNDERLINE],
       default: TerminalCursorStyle.BLOCK,
     },
+    [CodeTerminalSettingId.XtermRenderType]: {
+      type: 'string',
+      description: '%preference.terminal.integrated.xtermRenderTypeDesc%',
+      enum: ['webgl', 'canvas', 'dom'],
```
Comment (erha19) [inline packages/terminal-next/src/common/preference.ts:341]: 这里的值也可以替换掉

**Review by erha19 — CHANGES_REQUESTED** [review]

File: packages/terminal-next/src/common/preference.ts (hunk 2 above)
Comment (life2015) [reply packages/terminal-next/src/common/preference.ts]: 把 enum 抽到 common 了，简单重写了下

**Review by erha19 — DISMISSED** [review]

File: packages/terminal-next/src/common/preference.ts [hunk 3]
```diff
@@ -333,5 +338,11 @@ export const terminalPreferenceSchema: PreferenceSchema = {
       enum: [TerminalCursorStyle.BLOCK, TerminalCursorStyle.LINE, TerminalCursorStyle.UNDERLINE],
       default: TerminalCursorStyle.BLOCK,
     },
+    [CodeTerminalSettingId.XtermRenderType]: {
+      type: 'string',
+      description: '%preference.terminal.integrated.xtermRenderTypeDesc%',
+      enum: [RenderType.WebGL, RenderType.Canvas, RenderType.Dom],
+      default: 'webgl',
```
Comment (erha19) [inline packages/terminal-next/src/common/preference.ts:345]: 还是有漏网之鱼 🤔

**Review by erha19 — APPROVED** [review]

**Pull request merged**